File: gachisa-backend/src/main/java/com/gachisa/product/repository/ProductRepository.java
package com.gachisa.product.repository;

import com.gachisa.product.entity.Product;
import java.util.List;
import org.springframework.data.jpa.repository.JpaRepository;

public interface ProductRepository extends JpaRepository<Product, Long>, ProductRepositoryCustom {

    boolean existsBySellerIdAndName(Long sellerId, String name);

    boolean existsBySellerIdAndNameAndIdNot(Long sellerId, String name, Long id);

    List<Product> findBySellerId(Long sellerId);
}

File: gachisa-backend/src/main/kotlin/com/gachisa/product/repository/ProductRepository.kt
package com.gachisa.product.repository

import com.gachisa.product.entity.Product
import org.springframework.data.jpa.repository.JpaRepository

interface ProductRepository : JpaRepository<Product, Long>, ProductRepositoryCustom {

    fun existsBySellerIdAndName(sellerId: Long, name: String): Boolean

    fun existsBySellerIdAndNameAndIdNot(sellerId: Long, name: String, id: Long): Boolean

    fun findBySellerId(sellerId: Long): List<Product>
}
